decomposed/extensions/MDDE_LDM/Profile/Model/Methods/mdde_InitializeExportExtension.xml

```diff
@@ -0,0 +1,99 @@
+   applyTemplateOnObject obj, obj, attributeTemplateName, attributeExtendedAttributeName
+   
+   &#39;Process all child objects recursively in all packages
+   processPackage obj, obj, attributeTemplateName, attributeExtendedAttributeName
+   WriteInfo &quot;MDDE LDM Export extension initializion finished&quot;
+End Sub
+
+&#39; Check if the GTL output contains errors
+Function GTLOutputHasErrors(GTLResult)
+   GTLOutputHasErrors = False
+   If InStr(GTLResult, cMDDE_EXPORT_EXTENSION_CODE &amp; &quot;::Profile&quot;) Then
+      GTLOutputHasErrors = True
+   End If
+End Function
+
+Sub processPackage(objModel, objPackage, attributeTemplateName, attributeExtendedAttributeName)
+
+   &#39;Loop over Entities
+   Dim ent
+   For Each ent in objPackage.Entities
+      If Not ent.IsShortCut Then
+         applyTemplateOnObject objModel, ent, &quot;mddex_Entity_Export_Template&quot;, &quot;mddex_Entity_Export_Content&quot;
+         applyTemplateOnObject objModel, ent, attributeTemplateName, attributeExtendedAttributeName            
+      End If   
+   Next
+   
+   &#39;Loop over EntityAttributes
+   Dim entAttr
+   For Each entAttr in objPackage.EntityAttributes
+      If Not entAttr.IsShortCut Then
+         applyTemplateOnObject objModel, entAttr, &quot;mddex_EntityAttribute_Export_Template&quot;, &quot;mddex_EntityAttribute_Export_Content&quot;
+         applyTemplateOnObject objModel, entAttr, attributeTemplateName, attributeExtendedAttributeName            
+      End If   
+   Next
+   
+   &#39;Loop over Mappings
+   Dim map
+   For Each map in objPackage.Mappings
+       If Not map.IsShortCut Then
+         applyTemplateOnObject objModel, map, &quot;mddex_BaseClassifierMapping_Export_Template&quot;, &quot;mddex_BaseClassifierMapping_Export_Content&quot;
+         applyTemplateOnObject objModel, map, attributeTemplateName, attributeExtendedAttributeName            
+      End If   
+   Next
+   
+   &#39;Loop over AttributeMappings
+   Dim attrMap
+   For Each attrMap in objPackage.ModelEntityAttributeMappings
+      If Not attrMap.IsShortCut Then
+         applyTemplateOnObject objModel, attrMap, &quot;mddex_BaseStructuralFeatureMapping_Export_Template&quot;, &quot;mddex_BaseStructuralFeatureMapping_Export_Content&quot;         
+      End If   
+   Next
+   
+   &#39;Loop over packages
+   Dim childPackage
+   For Each childPackage in objPackage.Packages
+      processPackage objModel, childPackage, attributeTemplateName, attributeExtendedAttributeName
+   Next
+   
+End Sub
+
+&#39;Apply template on object and check for translation errors
+Sub applyTemplateOnObject(objModel, objItem, templateName, extendedAttributeName)
+   &#39;Process the object specific template
+   Dim evaluatedTemplate: evaluatedTemplate = objItem.EvaluateTemplateFor(templateName, cMDDE_EXPORT_EXTENSION_CODE)
+   &#39;Clear the extended attribute if it is set
+   If Not objItem.getExtendedAttributeText(extendedAttributeName) = &quot;&quot; Then
```

Comment:
I think it would also be more readable if you would do an Else statement on line 97 (as part of the other IF) to set the extended attribute to Nothing if the template returned nothing.

Reply:
I don't feel <> is that much more readable but changed it anyway :-) 
Best to re-review the initialize method because i have added additional error handling